parser/src/test/java/com/facebook/coresql/parser/sqllogictest/java/SqlLogicTest.java

```diff
@@ -0,0 +1,216 @@
+import static java.nio.file.FileVisitResult.CONTINUE;
+import static java.nio.file.Files.newBufferedReader;
+import static java.nio.file.Files.walkFileTree;
+
+public final class SqlLogicTest
+{
+    private SqlLogicTestConfig config;
+    private SqlLogicTestResult result;
+
+    private Set<AcceptedRdbms> skipIfSet;
+    private AcceptedRdbms onlyIf;
+    private FileVisitor<Path> fileVisitor;
+
+    protected enum SqlLogicTestError
+    {
+        PARSING_ERROR,
+        UNPARSING_ERROR,
+        UNPARSED_DOES_NOT_MATCH_ORIGINAL_ERROR
+    }
+
+    protected enum AcceptedRdbms
+    {
+        ALL,
+        ORACLE,
+        MYSQL,
+        MSSQL,
+        SQLITE,
+        POSTGRESQL
+    }
+
+    public SqlLogicTest() {}
+
+    public void run()
+            throws IOException
+    {
+        setup();
+        walkFileTree(Paths.get(TEST_FILES_ROOT_PATH), fileVisitor);
+        writeTestResultToFile(result);
+    }
+
+    private void setup()
+            throws IOException
+    {
+        config = new SqlLogicTestConfig();
+        result = new SqlLogicTestResult(config.getRdbms());
+        fileVisitor = new SqlLogicTestFileVisitor();
+        skipIfSet = new HashSet<>();
+        createFileForResults();
+    }
+
+    private void validateParsingOfSqlStatement(String statement)
+    {
+        SqlLogicTestError error = validateStatementAgainstPossibleErrors(statement);
+        if (error == null) {
+            result.incrementNumStatementsParsed();
+        }
+        else {
+            result.addErroneousStatement(new ErroneousSqlStatement(statement, error));
+        }
+    }
+
+    private class SqlLogicTestFileVisitor
+            implements FileVisitor<Path>
+    {
+        private void processStatement(BufferedReader reader)
+                throws IOException
+        {
+            StringBuilder statement = new StringBuilder();
+            while (true) {
+                final String line = reader.readLine();
+                if (line == null || isLineAfterStatement(line)) {
+                    break;
+                }
+                statement.append(line);
+            }
+            if (statement.length() > 0 && !statement.substring(statement.length() - 1).equals(";")) {
+                statement.append(";"); // NOTE: This semi-colon is necessary for parsing.
+            }
+            result.incrementNumStatementsFound();
+            validateParsingOfSqlStatement(statement.toString());
+        }
+
+        private boolean isLineBeforeStatement(String line)
+        {
+            return line.startsWith("statement ok") || line.startsWith("query");
+        }
+
+        private boolean statementIsAcceptableToCurrRdms()
+        {
+            if (config.getRdbms() == ALL) {
+                return skipIfSet.isEmpty() && onlyIf == null;
+            }
+
+            if (onlyIf != null && onlyIf != config.getRdbms()) {
+                return false;
+            }
+            return !skipIfSet.contains(config.getRdbms());
```

Comment:
nit: This is the fourth time I'm seeing config.getRdbms() - maybe pull this out into a member variable

Reply:
We now load all relevant fields from SqlLogicTestConfig into member variables (including the field mentioned here).

Done.